cmd/assets.go

```diff
@@ -0,0 +1,205 @@
+package cmd
+
+import (
+	"fmt"
+
+	compassv1beta1 "github.com/odpf/compass/api/proto/odpf/compass/v1beta1"
+	"github.com/odpf/salt/printer"
+	"github.com/odpf/salt/term"
+
+	"github.com/MakeNowJust/heredoc"
+	"github.com/spf13/cobra"
+)
+
+func assetsCommand() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:     "assets",
+		Aliases: []string{"assets"},
+		Short:   "Manage assets",
+		Annotations: map[string]string{
+			"group:core": "true",
+		},
+		Example: heredoc.Doc(`
+			$ compass assets list
+			$ compass assets get
+			$ compass assets delete
+			$ compass assets post
+		`),
+	}
+
+	cmd.AddCommand(listAllAssetsCommand())
+	cmd.AddCommand(getAssetByIDCommand())
+	cmd.AddCommand(postAssetCommand())
+	cmd.AddCommand(deleteAssetByIDCommand())
+
+	return cmd
+}
+
+func listAllAssetsCommand() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "list",
+		Short: "lists all assets",
+		Example: heredoc.Doc(`
+			$ compass assets list --host=<hostaddress> --header=<key>:<value>
+		`),
+		Annotations: map[string]string{
+			"action:core": "true",
+		},
+		RunE: func(cmd *cobra.Command, args []string) error {
+			spinner := printer.Spin("")
+			defer spinner.Stop()
+			cs := term.NewColorScheme()
+			fmt.Println("header:", header)
+			client, cancel, err := createClient(cmd, host)
+			if err != nil {
+				return err
+			}
+			defer cancel()
+
+			ctx := setCtxHeader(cmd.Context(), header)
+			res, err := client.GetAllAssets(ctx, &compassv1beta1.GetAllAssetsRequest{})
+			if err != nil {
+				return err
+			}
+
+			fmt.Println(cs.Bluef(prettyPrint(res.GetData())))
```

Comment:
instead of by default printing it as json, should we print it in a table format

example:
```
URN           NAME             TYPE        SERVICE
some-urn      some-name        table       bigquery
```
we still also provide an alternative to output as json tho, but by default we can start with table first.

wdyt? @ravisuhag

Reply:
I can agree that the table is usually more presentable, but here since we are developing a cli that supports us with making requests to compass, I think printing it in `json` should be fine, since it is usually how most of the responses. Have tried to keep the output presentable on our part. 

Also, I think if we represent this response in the table it will be a bit difficult to maintain all the key terms in a column. for e.g. for the below response:

```
{
		"id": "dffbe72e-96e6-4c1e-8752-6bee19c2fa11",
		"urn": "myjob::tableb",
		"type": "job",
		"service": "myjob",
		"name": "ironman_extractor",
		"version": "0.1",
		"updated_by": {
			"uuid": "stewart.jingga@gojek.com"
		},
		"created_at": {
			"seconds": 1652932353,
			"nanos": 731728000
		},
		"updated_at": {
			"seconds": 1652932353,
			"nanos": 731728000
		}
	}
```

I feel that things like updated_by, and assets array in discussion will tough to present in table. Also If we add more fields in response in an asset later, we will need to update cmd/assets.go as well.

Reply:
We can do it like how k8s does. If we are outputing json format as default, imho it will be too overwhelming and close to not being usable unless it is being piped to a file. Not to mention we have lots of record.

With table, we only show relevant information and as a quick look on the assets. We will still allow output to be in json format with a flag e.g. `-o json`.

Reply:
Have added this, take a look once.

<img width="789" alt="Screenshot 2022-05-23 at 8 18 31 AM" src="https://user-images.githubusercontent.com/57063469/169733930-6f345559-6e5f-4dce-a7c1-47d529ec062e.png">